lib/timers.js

```diff
@@ -62,55 +110,72 @@ function listOnTimeout() {
   var now = Timer.now();
   debug('now: %s', now);
 
-  var diff, first, threw;
-  while (first = L.peek(list)) {
-    diff = now - first._idleStart;
+  var diff, timer, threw;
+  while (timer = L.peek(list)) {
+    diff = now - timer._idleStart;
+
+    // Check if this loop iteration is to early for the timer.
+    // This happens if one of two things is true:
+    // - There are more timers scheduled for later in the list. (Most common)
```

Comment:
`for later` in this case could mean _temporally_ later (as in, "later today"), or _spatially_ later (as in, "later in the array"), or both — it might be good to clarify which is meant here.